store/sqlstore/system_store.go

```diff
@@ -86,3 +88,37 @@ func (s SqlSystemStore) PermanentDeleteByName(name string) (*model.System, *mode
 
 	return &system, nil
 }
+
+// InsertIfExists inserts a given system value if it does not already exist. If a value
+// already exists, it returns the old one, else returns the new one.
+func (s SqlSystemStore) InsertIfExists(system *model.System) (*model.System, *model.AppError) {
```

Comment:
Should we add a test for this? Maybe spawning a few goroutines and make sure the results are consistent.

Reply:
You get full marks for maintaining "high standards". I like it :)